Kallisto/Kallisto-Index.cwl

```diff
@@ -1,53 +1,80 @@
 class: CommandLineTool
+label: Kallisto index
+doc: |2
 
-hints:
-  DockerRequirement:
-    dockerPull: quay.io/biocontainers/kallisto:0.45.0--hdcc98e5_0
-  SoftwareRequirement:
-    packages:
-      kallisto:
-        version: [ "0.45.0" ]
-        specs: [ https://identifiers.org/biotools/kallisto ]
+  Docs: https://pachterlab.github.io/kallisto/
+
+
+  Builds a kallisto index
+
+  Usage: kallisto index [arguments] FASTA-files
+
+  Required argument:
+  -i, --index=STRING          Filename for the kallisto index to be constructed 
+
+  Optional argument:
+  -k, --kmer-size=INT         k-mer (odd) length (default: 31, max value: 63)
+  -t, --threads=INT           Number of threads to use (default: 1)
+  -d, --d-list=STRING         Path to a FASTA-file containing sequences to mask from quantification
+      --make-unique           Replace repeated target names with unique names
+      --aa                    Generate index from a FASTA-file containing amino acid sequences
+      --distinguish           Generate index where sequences are distinguished by the sequence name
+  -T, --tmp=STRING            Temporary directory (default: tmp)
+  -m, --min-size=INT          Length of minimizers (default: automatically chosen)
+  -e, --ec-max-size=INT       Maximum number of targets in an equivalence class (default: no maximum)
 
 inputs:
-  InputFiles:
-    type: File[]
-    format: edam:format_1929 # FASTA
-    inputBinding:
-      position: 200
-    
   IndexName:
     type: string
     inputBinding:
-      prefix: "--index="
+      prefix: --index=
+      valueFrom: $(self).kallistoIndex
       separate: false
-      valueFrom: $(self).kl
-
-#Optional arguments
-
+  inputFasta:
+    type: File[]
+    format: edam:format_1929
+    inputBinding:
+      position: 200
   kmerSize:
     type: int?
     inputBinding:
-      prefix: "--kmer-size="
+      prefix: --kmer-size=
       separate: false
-
   makeUnique:
     type: boolean?
     inputBinding:
-      prefix: "--make-unique"
-
-baseCommand: [kallisto, index]
+      prefix: --make-unique
 
 outputs:
-
   index:
     type: File
     outputBinding:
-      glob: $(inputs.IndexName)
+      glob: $(inputs.IndexName).kallistoIndex
+
+baseCommand:
+- kallisto
+- index
+
+hints:
+  DockerRequirement:
+    dockerPull: quay.io/biocontainers/kallisto:0.51.1--ha4fb952_1
+  SoftwareRequirement:
+    packages:
+    - package: kallisto
+      specs:
+      - https://identifiers.org/rrid/RRID:SCR_016582
+      - https://identifiers.org/biotools/kallisto
+      version:
+      - 0.51.1
 
 $namespaces:
-  edam: http://edamontology.org/
+  edam: https://edamontology.org/
+  s: https://schema.org/
 $schemas:
-  - https://edamontology.org/EDAM_1.18.owl
+- https://edamontology.org/EDAM_1.25.owl
+- https://schema.org/version/latest/schemaorg-current-https.rdf
+s:citation: https://dx.doi.org/10.1038/nbt.3519
+s:codeRepository: https://github.com/pachterlab/kallisto
+s:license: https://spdx.org/licenses/BSD-2-Clause
```

Comment:
Hello @Brilator ; this annotations imply that the citation for this CWL description is at that DOI, etc..

If you would like to add extra citations about the tool itself, you could add them to `SoftwareRequirement.packages` entry. Personally I stop after adding any relevant bio.tools and/or RRID identifiers as they can be used to look up the rest of the relevant metadata.

Reply:
Hi @mr-c,
I see. Thanks. 
That also mostly answers this question https://cwl.discourse.group/t/best-practice-metadata-and-attribution/993